src/main/java/me/desht/pneumaticcraft/common/tubemodules/ModuleNetworkManager.java

```diff
@@ -60,7 +60,7 @@ private Set<AbstractTubeModule> computeConnections(AbstractTubeModule module) {
         while (!pendingPositions.isEmpty()) {
             BlockPos pos = pendingPositions.pop();
             PneumaticCraftUtils.getTileEntityAt(level, pos, PressureTubeBlockEntity.class).ifPresent(tube -> tube.tubeModules()
-                    .filter(tm -> tm instanceof INetworkedModule && module.getClass() == tm.getClass())
+                    .filter(tm -> tm instanceof INetworkedModule)
```

Comment:
Might be worth adding an `AbstractTubeModule#canConnectTo(AbstractTubeModule other)` method for this.  Default would be to check for the same class, but could be overridden for the thermostat module to also connect to redstone modules.

Reply:
That's sounds like a good idea.